### PR TITLE
[Attestation Service] MessageBird provider

### PR DESCRIPTION
**Pull request by timmoreton**
### Description

* Add new `messagebird` provider for [MessageBird](https://messagebird.com/en/)

### Other changes

* Providers can now specify whether their callback for delivery status is `GET` or `POST`
* Minor typos in the example config

### Tested

Locally. Sent messages via MessageBird, confirmed delivery status receipts for both success and failure conditions.

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/attestation-service/src/sms/index.ts [hunk 1]
```diff
@@ -148,7 +157,7 @@ export function configuredSmsProviders() {
 }
 
 export function smsProvidersWithDeliveryStatus() {
-  return smsProviders.filter((provider) => provider.supportsDeliveryStatus())
+  return smsProviders.filter((provider) => provider.deliveryStatusMethod())
```
Comment (aslawson) [inline packages/attestation-service/src/sms/index.ts:160]: what is the false state to this filter?  will an empty string result to false?

File: packages/attestation-service/src/sms/messagebird.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  deliveryStatusURL: string | undefined
+
+  constructor(apiKey: string, unsupportedRegionCodes: string[]) {
+    super()
+    this.messagebird = initMB(apiKey)
+    this.unsupportedRegionCodes = unsupportedRegionCodes
+  }
+
+  async receiveDeliveryStatusReport(req: express.Request, logger: Logger) {
+    if (typeof req.query.reference === 'string' && typeof req.query.status === 'string') {
```
Comment (aslawson) [inline packages/attestation-service/src/sms/messagebird.ts:31]: didn't realize you could parse a GET query fields this way with typescript. cool :)

File: packages/attestation-service/src/sms/messagebird.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      case 'sent':
+        return AttestationStatus.Upstream
+      case 'buffered':
+      case 'scheduled':
+        return AttestationStatus.Queued
+    }
+    return AttestationStatus.Other
+  }
+
+  deliveryStatusMethod = () => 'GET'
```
Comment (aslawson) [inline packages/attestation-service/src/sms/messagebird.ts:58]: weird that its a GET used to pass information...

**Review by aslawson — APPROVED** [review]

File: packages/attestation-service/src/sms/messagebird.ts (hunk 3 above)
Comment (timmoreton) [reply packages/attestation-service/src/sms/messagebird.ts:58]: yep, weird

**Pull request merged**